File: JAVA/PokerProtocol/src/bluffinmuffin/protocol/commands/game/PlayerJoinedCommand.java
package bluffinmuffin.protocol.commands.game;

import java.util.StringTokenizer;

import bluffinmuffin.protocol.commands.AbstractCommand;

public class PlayerJoinedCommand extends AbstractCommand
{
    @Override
    protected String getCommandName()
    {
        return PlayerJoinedCommand.COMMAND_NAME;
    }
    
    private final int m_playerPos;
    private final String m_playerName;
    private final int m_playerMoney;
    
    public static String COMMAND_NAME = "gamePLAYER_JOINED";
    
    public PlayerJoinedCommand(StringTokenizer argsToken)
    {
        m_playerPos = Integer.parseInt(argsToken.nextToken());
        m_playerName = argsToken.nextToken();
        m_playerMoney = Integer.parseInt(argsToken.nextToken());
    }
    
    public PlayerJoinedCommand(int pos, String name, int money)
    {
        m_playerPos = pos;
        m_playerName = name;
        m_playerMoney = money;
    }
    
    @Override
    public void encode(StringBuilder sb)
    {
        append(sb, m_playerPos);
        append(sb, m_playerName);
        append(sb, m_playerMoney);
    }
    
    public int getPlayerPos()
    {
        return m_playerPos;
    }
    
    public String getPlayerName()
    {
        return m_playerName;
    }
    
    public int getPlayerMoney()
    {
        return m_playerMoney;
    }
}

File: trunk/C#/PokerProtocol/Commands/Game/PlayerJoinedCommand.cs
using System.Text;
using EricUtility;
using EricUtility.Networking.Commands;

namespace PokerProtocol.Commands.Game
{
    public class PlayerJoinedCommand : AbstractTextCommand
    {
        public static string COMMAND_NAME = "gamePLAYER_JOINED";

        private readonly int m_PlayerPos;
        private readonly int m_PlayerMoney;
        private readonly string m_PlayerName;

        public int PlayerPos
        {
            get { return m_PlayerPos; }
        }
        public int PlayerMoney
        {
            get { return m_PlayerMoney; }
        }
        public string PlayerName
        {
            get { return m_PlayerName; }
        }

        public PlayerJoinedCommand(StringTokenizer argsToken)
        {
            m_PlayerPos = int.Parse(argsToken.NextToken());
            m_PlayerName = argsToken.NextToken();
            m_PlayerMoney = int.Parse(argsToken.NextToken());
        }

        public PlayerJoinedCommand(int pos, string name, int money)
        {
            m_PlayerPos = pos;
            m_PlayerName = name;
            m_PlayerMoney = money;
        }

        public override void Encode(StringBuilder sb)
        {
            Append(sb, m_PlayerPos);
            Append(sb, m_PlayerName);
            Append(sb, m_PlayerMoney);
        }
    }
}
